### ISSUE TITLE
After update to v1.4.0 "sudo: unknown uid 1553427339: who are you?"

### ISSUE THREAD

**Issue by shilgendorff** [drud/ddev#1281]
**Describe the bug**
i updated ddev to verison 1.4.0 it always displays this error `sudo: unknown uid xxxxxxxxx: who are you?`  when i  call `ddev start`.

ddev is installed with homebrew.

ddev show: Failed to start ddev: web container failed: log=, err=container exited, please use 'ddev logs -s web` to find out why it failed 
 
ddev logs -s web => 
```
+ DDEV_PHP_VERSION=7.2
+ DDEV_WEBSERVER_TYPE=nginx-fpm
+ '[' -n htdocs ']'
+ export WEBSERVER_DOCROOT=/var/www/html/htdocs
+ WEBSERVER_DOCROOT=/var/www/html/htdocs
+ export NGINX_DOCROOT=/var/www/html/htdocs
+ NGINX_DOCROOT=/var/www/html/htdocs
+ '[' -f /mnt/ddev_config/nginx-site.conf ']'
+ '[' -f /mnt/ddev_config/apache/apache-site.conf ']'
+ '[' -n 7.2 ']'
+ update-alternatives --set php /usr/bin/php7.2
+ ln -sf /usr/sbin/php-fpm7.2 /usr/sbin/php-fpm
+ export PHP_INI=/etc/php/7.2/fpm/php.ini
+ PHP_INI=/etc/php/7.2/fpm/php.ini
+ '[' -d /mnt/ddev_config/php ']'
+ '[' typo3 = backdrop ']'
+ rm -f /etc/nginx/nginx-site.conf
+ '[' -f /etc/nginx/nginx-site-typo3.conf ']'
+ ln -s /etc/nginx/nginx-site-typo3.conf /etc/nginx/nginx-site.conf
+ rm -f /etc/apache2/apache-site.conf
+ '[' -f /etc/apache2/ddev_apache-typo3.conf ']'
+ ln -s -f /etc/apache2/apache-site-default.conf /etc/apache2/apache-site.conf
+ envsubst '$WEBSERVER_DOCROOT,$NGINX_DOCROOT'
+ envsubst '$WEBSERVER_DOCROOT'
++ id -u
++ id -g
+ printf '\nexport APACHE_RUN_USER=uid_1553427339\nexport APACHE_RUN_GROUP=gid_637725509\n'
+ '[' nginx-fpm = apache-cgi ']'
+ '[' nginx-fpm = apache-fpm ']'
+ '[' false '!=' true ']'
+ disable_xdebug
Disabled xdebug
+ ls /var/www/html
++ id -u
++ id -g
+ sudo chown -R 1553427339:637725509 /mnt/composer_cache/
sudo: unknown uid 1553427339: who are you?
```

**Version and configuration information (please complete the following information):**
 - Host computer OS and Version: macOS Mojave]
 - Docker version:
```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:31 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:29:02 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

 - ddev version : 
```
dba           	drud/phpmyadmin:v1.4.0    
router        	drud/ddev-router:v1.4.0   
domain        	ddev.local                
cli           	v1.4.0                    
web           	drud/ddev-webserver:v1.4.0
db            	drud/ddev-dbserver:v1.4.0 
ddev-ssh-agent	drud/ddev-ssh-agent:v1.4.0
commit        	v1.4.0
```
 - No custom configurations

**Additional context**
Add any other context about the problem here. Thanks!

**Comment by rfay** [comment]
Thanks for your care in writing this very clear issue.

The problem is obviously `sudo: unknown uid 1553427339: who are you?`  1553427339 is way outside the range of uids for Linux (and macOS). 

Do you have any customization at all in your .ddev directory, like changes to the config.yaml or docker-compose.*.yaml? If so, please back those out for starters.

On your mac, what does `id` tell you?

Thanks!

**Comment by rfay** [comment]
I assume you're on macOS Mojave? It seems the range for UIDs on Mojave was somehow increased over the traditional. I'll have a workaround soon. We've always had to adapt on this problem with Windows, not sure why there's be a difference in v1.4.0.

**Comment by rfay** [comment]
Please try this temporary workaround, I think it will work for you. 

Add a `docker-compose.uid.yaml` in your project .ddev directory with these contents:

```
version: '3.6'
services:
  web:
    user: "1001:1001"
  db:
    user: "1001:1001"
```

And then `ddev start`. That will make the containers run as 1001 instead of your outrageous Mojave uid :) We'll get this solved in v1.5.0 or sooner. I do with I knew how and when these huge UIDs started on macOS.

**Comment by shilgendorff** [comment]
ohw wauww, thanks a lot it works now - and if you know the workaround, it's easy :-)  🥇

**Comment by rfay** [comment]
Reopening, we'll need to solve this.

**Comment by rfay** [comment]
The immediate problem here is https://github.com/drud/ddev/blob/e4726e05a277a4702556001f5e63ced058ce24e9/pkg/ddevapp/ddevapp.go#L846-L851 

This stanza actually used to change the uid/gid if they were >60000, but no more; now (assuming this problem only occurred on Windows) we use an arbitrary UID only if the uid is non-numeric. See https://github.com/drud/ddev/blob/1893afc3dede578ec1de06473ff972ff225862a9/pkg/util/utils.go#L147-L168

Also, the message about running as root is incorrect and should have been gotten rid of.

**Comment by rfay** [comment]
For everyone affected by this bug, we have a fix for this in https://github.com/drud/ddev/pull/1285 and are considering a point release. Could you please test the version downloadable from https://circleci.com/gh/drud/ddev/6007#artifacts/containers/0 ? @shilgendorff and any others.

**Comment by milesbigbluedoor** [comment]
@rfay Tested: Confirmed I am now able to `ddev start` container and `ddev ssh` into the container. Once in the container `id` correctly returns `uid=1000(uid_1000) gid=1000(gid_1000) groups=1000(gid_1000)`. Thank you very much for getting this issue resolved so quickly.

**Comment by milesbigbluedoor** [comment]
@shilgendorff I would recommend just doing a quick test within the download folder for this version:

Within the download folder execute `./ddev version` to verify you are utilising that version of ddev.

From there create a new directory within that folder, for example: 
```
mkdir ddev-fix-test
```

Change to this directory and setup a simple php container with ddev, start that container, then enter the container and run `id` to check if the container is using the correct uid:

```
cd ddev-fix-test
../ddev config --project-type php --php-version 7.1
../ddev start
../ddev ssh
id
```
Should print: `uid=1000(uid_1000) gid=1000(gid_1000) groups=1000(gid_1000)`

As for working with the fix, I would either swap back to 1.3.0 for now by doing the following:
```
brew list --versions ddev
brew switich ddev 1.3.0
```

or you could move the fix to your working directory and execute your ddev commands like above we did above for the test
for example:
```
./ddev config
./ddev start
./ddev sequelpro
```
etc.

Hope this helps.

**Comment by rfay** [comment]
We released v1.4.1 with this fix in it just now, https://github.com/drud/ddev/releases/tag/v1.4.1

If you use homebrew, you can `brew upgrade ddev`

Thanks all for your help.
